airgun/views/repository.py

```diff
@@ -122,6 +122,28 @@ def is_displayed(self):
         )
 
 
+class AuthorizationEntry(EditableEntry):
+
+    clear_button = Text(
+        ".//span[contains(@ng-hide, 'editMode')]/i[@ng-show='deletable']")
+
+    @View.nested
+    class edit_field(View):
+        username = TextInput(id='upstream_username')
+        password = TextInput(id='upstream_password')
+
+    def fill(self, value):
+        """Handle the clear functionality, if username is supplied with None or empty
```

Comment:
Wouldn't it be better to update EditableEntry's fill() instead? (if None or empty string - hit 'clear' button)?